ot3_state_manager/ot3_state_manager/ot3_state_manager.py

```diff
@@ -1,213 +1,118 @@
-
-        If hardware is not attached for axis, then position value will be None.
-        """
-        pos_dict = self._pos_dict()
-
-        # Ignoring the mypy error because we are checking that the value is None
-        return {
-            key: (pos_dict[key].current_position if value is not None else None)  # type: ignore[union-attr]
-            for key, value in pos_dict.items()
-        }
-
-    @property
-    def commanded_position(self) -> Dict[OT3Axis, Optional[float]]:
-        """Returns dictionary with axis mapped to commanded position of said axis.
-
-        If hardware is not attached for axis, then position value will be None.
+            host: hostname to connect to
+            port: port to connect to
+            ot3_state: OT3State object to modify
         """
-        pos_dict = self._pos_dict()
-
-        # Ignoring the mypy error because we are checking that the value is None
-        return {
-            key: (pos_dict[key].commanded_position if value is not None else None)  # type: ignore[union-attr]
-            for key, value in pos_dict.items()
-        }
-
-    @property
-    def encoder_position(self) -> Dict[OT3Axis, Optional[float]]:
-        """Returns dictionary with axis mapped to encoder position of said axis.
+        self._host = host
+        self._port = port
+        self._ot3_state = ot3_state
 
-        If hardware is not attached for axis, then position value will be None.
-        """
-        pos_dict = self._pos_dict()
-
-        # Ignoring the mypy error because we are checking that the value is None
-        return {
-            key: (pos_dict[key].encoder_position if value is not None else None)  # type: ignore[union-attr]
-            for key, value in pos_dict.items()
-        }
-
-    @property
-    def updatable_axes(self) -> List[OT3Axis]:
-        """Returns a list of positions that are available to be updated."""
-        return [key for key, value in self._pos_dict().items() if value is not None]
-
-    @property
-    def is_gripper_attached(self) -> bool:
-        """Returns True if gripper is attached, False if not."""
-        return self.gripper is not None
-
-    @property
-    def is_left_pipette_attached(self) -> bool:
-        """Returns True if left pipette is attached, False if not."""
-        return self.left_pipette is not None
-
-    @property
-    def is_right_pipette_attached(self) -> bool:
-        """Returns True if right pipette is attached, False if not."""
-        return self.right_pipette is not None
-
-    @property
-    def is_high_throughput_pipette_attached(self) -> bool:
-        """Returns True if high throughput pipette is attached, False if not."""
-        return (
-            self.right_pipette is not None
-            and self.right_pipette.model == PipetteModel.MULTI_96_1000
+    async def _handle_message(
+        self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter
```

Comment:
I think we don't want to use tcp and streams here but rather udp and datagrams: https://docs.python.org/3/library/asyncio-protocol.html#udp-echo-client

Reply:
we're sending discrete messages after all, and we'd like to be able to loadshed if we need to, so udp is a much better fit

Reply:
https://github.com/Opentrons/ot3-firmware/pull/429/commits/ef93b42b7f6b7e14ff0d6f997de22f4e894695ae

Here is my first pass at implementing this.